### PR TITLE
[WIP] Madison/custom apis

### PR DESCRIPTION
**Pull request by madisonmay**

### CODE REVIEW

File: IndicoIo/IndicoIo.php [hunk 1]
```diff
@@ -378,6 +239,80 @@ static function get(&$array, $key) {
...
+     		$params['batch'] = False;
+     		try {
+     			$data[0] = Image::processImage($data[0], 144, true);
+     		} catch (Exception $e) {}
+     	} else {
+     		$params['batch'] = True;
+     		try {
+     			$x = array();
+     			$y = array();
+     			foreach ($arr as $pair) {
```
Comment (sihrc) [inline IndicoIo/IndicoIo.php]: where is $arr defined? I think I'm misunderstanding something.

File: IndicoIo/IndicoIo.php [hunk 2]
```diff
@@ -378,6 +239,80 @@ static function get(&$array, $key) {
...
+     		} catch (Exception $e) {}
+     	}
+     	return IndicoIo::_callService($data, 'custom', 'add_data', $params);
+    }
+
+    function predict($data, $params=array()) {
+    	$params['collection'] = $this->name;
+    	try {
+    		$data = Image::processImage($data, 144, true);	
+    	} catch (Exception $e) {}
```
Comment (sihrc) [inline IndicoIo/IndicoIo.php]: Is htis our way of handling not images passed in?

File: IndicoIo/IndicoIo.php (hunk 1 above)
Comment (madisonmay) [inline IndicoIo/IndicoIo.php]: No -- that's just invalid code.The exception handling needs to be more explicit about error type (is the secondary issue here).

File: IndicoIo/IndicoIo.php (hunk 2 above)
Comment (madisonmay) [inline IndicoIo/IndicoIo.php]: Roughly -- again I need to except on the proper error types.

**Comment by madisonmay** [comment]
Closing and reopening into development.

**Pull request closed without merging**